### DISCUSSION TITLE
Prevent update erasing database?

### DISCUSSION THREAD

**Discussion by odonnellmj** [MusicPlayerDaemon/MPD#1378] [Q&A]
I have 2.5TB of music on a USB mounted disk. I have a problem that spontaneously changes permissions, so that the mpd user cannot access the music library hierarchy. That is not an MPD problem, but there might be some MPD configuration to ameliorate it.

If I fail to notice the permissions problem, and issue "mpc --wait update" I get a very quick update which erases the database. Then, after fixing the permissions problem, I have  a rather long update to rebuild the whole thing.

I have hunted for a configuration option for mpd or mpc to give an error message for inaccessible music directory, instead of erasing the database. There doesn't appear to be one, and I can see a reason why it isn't trivial to provide (I might have deleted a subdirectory with the intention of removing the music files in it). It probably should be based on a distinction between things that are totally removed vs. inaccessible.

I expect that the same problem will arise if USB storage is accidentally disconnected or unmounted.

So, I am not optimistic, but if someone knows a way to make the configuration safer, that will be very helpful, and this could be a useful improvement for a later version, since intermittent connection to storage is probably a common thing with large music libraries.

**Comment by coroner21** [comment]
Why not simply disable automatic database updates?

**Reply by odonnellmj** [reply]
coroner21:

    Why not simply disable automatic database updates?

I never enabled automatic database updates. This problem occurs when I cause an update with "mpc --wait update". I will try to remember to check the permissions of the music directory before doing so, but it would be better to be less vulnerable to an error. Also, the spontaneous change in permissions could come with an update in progress, and as far as I know, other accidents that disturb access to the files, such as a dislodged USB cable.
